src/gennylib/include/gennylib/context.hpp

```diff
@@ -158,6 +158,9 @@ class ExternalPhaseCoordinator {
     const int m_socket;
 };
 
+// The compiler is not happy if we don't forward-declare it here; see impl below
```

Comment:
Noting that this comment and "strange-ness" has not been added in this PR, it's just been moved up